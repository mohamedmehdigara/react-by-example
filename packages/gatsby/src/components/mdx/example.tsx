import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
