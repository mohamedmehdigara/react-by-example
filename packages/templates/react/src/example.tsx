import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
