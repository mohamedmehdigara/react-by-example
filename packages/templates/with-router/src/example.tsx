import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
