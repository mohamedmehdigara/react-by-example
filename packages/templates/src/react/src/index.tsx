import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
