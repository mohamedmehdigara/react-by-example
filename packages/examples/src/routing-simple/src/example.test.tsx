import { cleanup, render, RenderResult } from '@testing-library/react'
import React, { FC } from 'react'
import { MemoryRouter } from 'react-router'
import { Simple } from './example'

describe('Simple', () => {
  let component: RenderResult

  interface TestProps {
    location: string
  }
  const Test: FC<TestProps> = ({ location }) => {
    return (
      <MemoryRouter initialEntries={[location]}>
        <Simple />
      </MemoryRouter>
    )
  }

  afterEach(() => {
    cleanup()
  })

  describe('when home', () => {
    beforeEach(() => {
      component = render(<Test location="/" />)
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('when lorem', () => {
    beforeEach(() => {
      component = render(<Test location="/lorem" />)
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('when ipsum', () => {
    beforeEach(() => {
      component = render(<Test location="/ipsum" />)
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('when dolor', () => {
    beforeEach(() => {
      component = render(<Test location="/dolor" />)
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })

  describe('when 404', () => {
    beforeEach(() => {
      component = render(<Test location="/404" />)
    })

    it('should snapshot', () => {
      expect(component.container).toMatchSnapshot()
    })
  })
})
