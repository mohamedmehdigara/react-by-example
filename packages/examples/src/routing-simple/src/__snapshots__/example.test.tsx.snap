// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple when 404 should snapshot 1`] = `
<div>
  <h1>
    Home
  </h1>
  <ul>
    <li>
      <a
        href="/lorem"
      >
        lorem
      </a>
    </li>
    <li>
      <a
        href="/ipsum"
      >
        ipsum
      </a>
    </li>
    <li>
      <a
        href="/dolor"
      >
        dolor
      </a>
    </li>
  </ul>
</div>
`;

exports[`Simple when dolor should snapshot 1`] = `
<div>
  dolor
</div>
`;

exports[`Simple when home should snapshot 1`] = `
<div>
  <h1>
    Home
  </h1>
  <ul>
    <li>
      <a
        href="/lorem"
      >
        lorem
      </a>
    </li>
    <li>
      <a
        href="/ipsum"
      >
        ipsum
      </a>
    </li>
    <li>
      <a
        href="/dolor"
      >
        dolor
      </a>
    </li>
  </ul>
</div>
`;

exports[`Simple when ipsum should snapshot 1`] = `
<div>
  ipsum
</div>
`;

exports[`Simple when lorem should snapshot 1`] = `
<div>
  lorem
</div>
`;
